app/clients/hubee_api_client.rb

```diff
@@ -0,0 +1,80 @@
+class HubEEAPIClient < AbstractHubEEAPIClient
+  class NotFound < StandardError; end
+  class AlreadyExists < StandardError; end
+
+  def find_organization(organization)
+    http_connection.get("#{host}/referential/v1/organizations/SI-#{organization.siret}-#{organization.code_commune_etablissement}").body
+  rescue Faraday::ResourceNotFound
+    raise NotFound
+  end
+
+  def create_organization(organization, email)
+    http_connection.post(
+      "#{host}/referential/v1/organizations",
+      {
+        type: 'SI',
+        companyRegister: organization.siret,
+        branchCode: organization.code_commune_etablissement,
+        email:,
+        name: organization.denomination,
+        postalCode: organization.code_postal_etablissement,
+        territory: organization.code_commune_etablissement,
+        status: 'Actif'
+      }.to_json,
+      'Content-Type' => 'application/json'
+    )
+  rescue Faraday::BadRequestError => e
+    raise AlreadyExists if already_exists_error?(e)
+
+    raise
+  end
+
+  def create_subscription(authorization_request, organization_payload, process_code)
+    http_connection.post(
+      "#{host}/referential/v1/subscriptions",
+      {
+        datapassId: authorization_request.external_id.to_i,
+        notificationFrequency: 'unitaire',
+        processCode: process_code,
+        subscriber: {
+          type: 'SI',
+          companyRegister: organization_payload['companyRegister'],
+          branchCode: organization_payload['branchCode']
+        },
+        email: authorization_request.demandeur.email,
+        status: 'Actif',
+        localAdministrator: {
+          email: authorization_request.demandeur.email
+        },
+        delegationActor: {
+          branchCode: organization_payload['branchCode'],
+          companyRegister: organization_payload['companyRegister'],
+          type: 'EDT'
```

Comment:
c'est quoi ça, un param "EDiTeur" ?

comment on indique à HubEE l'éditeur de la commune (et qu'il faut utiliser le mode "API" au lieu de "portail" quand il y a un éditeur) ?

Reply:
Aucune idée, j'ai c/c la logique de DataPass.

Reply:
Après il y a un swagger pour les endpoints subscriptions, pas organisations. J'ai tout de même tapé sur les endpoints avec des valeurs pour vérifier les paramètres obligatoires and cie.

Par ailleurs, je pense que je vais faire le choix de ne pas coder de tests: j'ai développé un client spécifique à notre couche modèle, donc en réalité si je fais des tests je vais juste copier/coller le code fonctionnel et le mettre dans le test, je ne suis pas sûr que cela soit intéressant.